cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -1408,8 +1026,12 @@ Slice::JavaGenerator::writeMarshalUnmarshalCode(
                     return;
                 }
             }
+            // Check if either 1) No type metadata was applied to this sequence at all or 2) 'java:type' was applied to
+            // where the sequence is used (which overrides any metadata on the definition) or 3) 'java:type' was applied
+            // to the sequence definition, and there is no metadata overriding it where the sequence is used.
```

Comment:
An unrelated bug I stumbled across. We weren't always checking for these metadata in the same order, which could cause us to generate uncompilable code(because the types wouldn't match). Pretty niche corner case, but still.

You'll see me messing around with `java:type`, `java:buffer`, and `java:serializable` a few times in this file.
All I'm doing is making sure we check them in a consistent (and correct) order.